clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -709,7 +665,49 @@ private void discard() {
         }
     }
 
-    private class FetchManagerMetrics {
+    private static class FetchCompletion {
```

Comment:
Nitpick: would `CompletedFetch` be a better name? You seem to use that as the variable name in some places.